graph/middleware/knowledge.py

```
@@ -2,49 +2,192 @@
+                        msg.content
+                        if isinstance(msg.content, str)
+                        else str(msg.content)
+                    )
+                    break
+
+            if last_human:
+                results = self._store.search(last_human, k=self._top_k)
+                if results:
+                    context_parts = ["[Relevant knowledge from previous sessions:]"]
+                    for r in results:
+                        context_parts.append(f"- [{r['table']}] {r['preview']}")
```

Ruff (0.15.10)

[warning] Use `list.extend` to create a transformed list

(PERF401)